UPGRADE.md

```diff
@@ -8,6 +8,14 @@ you are using a URL-style DSN and have a username, password or database name con
 percentage sign, you need to update your DSN. If your password is, say, ``foo%foo``, it
 should be encoded as ``foo%25foo``.
 
+## MINOR BC BREAK: Native JSON type support in MySQL 5.7 platform
+
+MySQL 5.7 ships with native JSON type support which now is adopted by DBAL.
+If you are using MySQL 5.7 and have columns with ``json_array`` type defined in your database,
+utilizing DBAL's schema comparator will cause a schema diff, where the previously used native type
+``LONGTEXT`` will be changed to ``JSON``.
```

Comment:
Well, it will alter to a native `JSON` type later if the user changes their column definition some other way, which might be a bit unexpected (buggy?) but I guess technically not a BC break.

When the column is updated though it's actually quite likely to fail in practice because the `LONGTEXT` implementation used an empty string as a default value, which is invalid for native `JSON`. It's not just the case that it'll only fail if the user was doing something weird.

Not a blocker IMO - just making sure you're aware.